packages/flutter_tools/lib/src/flx.dart

```diff
@@ -115,7 +115,13 @@ Map<_Asset, List<_Asset>> _parseAssets(
 
       List<FileSystemEntity> files = assetDir.listSync(recursive: true);
 
+      final String buildDirPath = path.absolute(path.join(assetBase, 'build'));
```

Comment:
I see the referenced issue now. Yeah, I think we should disallow assets at the project root; even with build/ excluded, the extra tree searching is inefficient.